module-router.c

```diff
@@ -636,10 +636,19 @@ static void replace_whitespace_with_hash(char* string) {
  * @ret bool: true if name is valid and vice versa.
  */
 static bool is_valid_name(const char* name ) {
+	ROUTER_FUNCTION_ENTRY;
     bool is_valid = true;
-    if((name == NULL) || (name[0] == '\0')) {
-        is_valid = false;
+    if((name == NULL)) {
+        return false;
     }
+    if(name != NULL)
+    {
```

Comment:
This condition looks not necessary because "name" is already checked just above.